proto/jobworker/v1/service_api.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+
+package jobworker.v1;
+
+option go_package = "github.com/tjper/teleport/proto/jobworker/v1";
+
+// JobWorkerService provides methods that allow clients to start and interact
+// with arbitrary commands running on the service's host.
+service JobWorkerService {
+  rpc Start(StartRequest) returns (StartResponse){}
+  rpc Stop(StopRequest) returns (StopResponse){}
+  rpc Status(StatusRequest) returns (StatusResponse){}
+  rpc Output(OutputRequest) returns (stream OutputResponse){}
+}
+
+// StartRequest specifies job details for JobWorkerService.Start.
+message StartRequest {
+  // command is the arbitrary command to execute on JobWorkerService's host.
+  string command  = 1;
+  // limits are the resource limits to enforce on the job.
+  Limits limits   = 2;
+}
+
+// StartResponse informs clients started job details.
+message StartResponse {
+  string job_id  = 1;
+  // command is the arbitrary command that has been executed.
+  string command = 2;
+  // status is the human-readable current status of the job.
+  string status  = 3;
+  // limits are the resource limits being enforced on the job.
+  Limits limits  = 4;
+}
+
+// StopRequest specifies a job ID to stop for JobWorkerService.Stop.
+message StopRequest {
+  string job_id = 1;
+}
+
+// StopResponse is a placeholder. This will maintain backwards compatibility
+// in the event response details exist in the future.
+message StopResponse {}
+
+// StatusRequest specifies a job ID to perform a status check on for
+// JobworkerService.Status.
+message StatusRequest {
+  string job_id = 1;
+}
+
+// StatusResponse informs clients the status of a job.
+message StatusResponse {
+  // status is the current job state.
+  string status = 1;
```

Comment:
What are the potential values for status? Perhaps an enum would be better suited here?

Reply:
The different statuses are:
- running
- stopped
- exited w/ exit code

I would use an enum, but the `exited` status is dynamic. Above, I stated the status is human readable, meaning it's meant to be digested by people instead of programs. 

A couple choices I see here:
1. I create a new `Status` message that contains a `current` field that has an enum type and an exit-code field that is utilized when `current` is set to `exited`.
2. I don't report the exit code and change the `status` field to have an enum type.
3. Keep the current approach.

If I wasn't trying to limit the scope of this project, I'd go with 1.

Reply:
After revisiting this comment, expanding how status is reported is pretty simple so I've added approach 1 to the proto definition.